doc/manual.rst

```diff
@@ -4611,8 +4611,8 @@ The concept types can be parametric just like the regular generic types:
     M.data[m * M.N + n] = v
 
   # Adapt the Matrix type to the concept's requirements
-  template Rows*(M: type Matrix): expr = M.M
-  template Cols*(M: type Matrix): expr = M.N
+  template Rows*(M: type Matrix): int = M.M
```

Comment:
If `expr` is replaced by `untyped` (as noted below, line 5040), why do we here replace it with `int`?

Reply:
In that example code, ``M.M`` and `` M.N`` is always ``int`` type, and ``Rows`` and ``Cols`` is compared with ``int`` value in AnyMatrix concept.
There are templates in ``system`` module that have specific return type:
```nim
template currentSourcePath(): string
template `[]`(s: string; i: int): char
template `^`(x: int): BackwardsIndex
template likely(val: bool): bool
```
I think when a template always returns int type, return type of that template can be written as int.
If return type of a template depends on it's ``untyped`` parameter, return type must be ``untyped``.